backend-project/small_eod/notifications/views.py

```diff
@@ -0,0 +1,67 @@
+from django.forms.models import model_to_dict
+from rest_framework.views import APIView
+
+
+class SendNotificationsMixin(APIView):
+    notified_users = None
+    ignored_fields = []
+    init_instance = None
+
+    def send_notifications(self, http_action, **kwargs):
+        self.kwargs["pk"] = self.kwargs.get("pk", None) or kwargs["data"].get(
+            "id", None
+        )
+        instance = None
+
+        if http_action in ["post", "put"]:
+            instance = self.get_object()
+        else:
+            instance = self.init_instance
+
+        if http_action == "put" and not self.has_instance_changed(
+            self.init_instance, instance
+        ):
+            return
+
+        notified_users = self.get_notified_uers(instance)
+        kwargs["actor"] = self.basename
+        kwargs["action"] = self.action
+        kwargs["instance"] = model_to_dict(instance)
```

Comment:
Dlaczego nie bezpośrednio:
```suggestion
        kwargs["instance"] = instance
```
?

Reply:
Nie byłem pewny czy ten template engine umie w obiekty czy tylko w dicty